lib/ansible/modules/cloud/azure/azure_rm_storageaccount_facts.py

```diff
@@ -409,13 +418,17 @@ def account_obj_to_dict(self, account_obj, blob_service_props=None):
                 queue=self.format_endpoint_dict(account_dict['name'], account_key[0], account_obj.primary_endpoints.queue, 'queue'),
                 table=self.format_endpoint_dict(account_dict['name'], account_key[0], account_obj.primary_endpoints.table, 'table')
             )
+            if account_key[0]:
```

Comment:
Move to the L416's dict construct is OK, no need another conditional check

Reply:
If we move it into the dict construct, there will be an empty field in the results like `key: ""` if `I(show_connection_string=false)`. There is a check in the function `self.format_endpoint_dict` to check whether the account_key is empty or not. If it is empty, the results will not contain the `connection_string`. So, is it ok to move it into the dict construction?

Reply:
IMO, we can move the `get_connection_string` and construct the endpoints to one conditional check `if self.show_connection_string`.